config-ui/src/components/Nav.jsx

```diff
@@ -15,73 +15,104 @@
  * limitations under the License.
  *
  */
-import React from 'react'
-import { Alignment, Position, Popover, Navbar, Icon } from '@blueprintjs/core'
+import React, { useContext, useEffect, useState } from 'react'
+import { Position, Popover, Navbar, Icon } from '@blueprintjs/core'
 import '@/styles/nav.scss'
 import { ReactComponent as SlackIcon } from '@/images/slack-mark-monochrome-black.svg'
 import { ReactComponent as SlackLogo } from '@/images/slack-rgb.svg'
+import UIContext from '@/store/UIContext'
+import useWindowSize from '@/hooks/useWIndowSize'
 
 const Nav = () => {
+  const uiContext = useContext(UIContext)
```

Comment:
@henit-chobisa Does it help the situation any if you used De-structuring here?
```suggestion
  const { changeSidebarVisibility, desktopBreakpointWidth, sidebarVisible } = useContext(UIContext)
```

Then instead of referencing the Entire `UIContext` object, we can just utilize the De-structured children props?

Reply:
Yes Julien, it actually resolves the issue.
No side-effects and no linter issues, 
Thank you so much!!!